org.scala-ide.sdt.core/src/org/scalaide/core/internal/extensions/autoedits/ApplyTemplateCreator.scala

```diff
@@ -0,0 +1,13 @@
+package org.scalaide.core.internal.extensions.autoedits
+
+import org.scalaide.core.text.Document
+import org.scalaide.core.text.TextChange
+import org.scalaide.extensions.autoedits.ApplyTemplate
+
+object ApplyTemplateCreator {
+  def create(doc: Document, change: TextChange): ApplyTemplate =
+    new ApplyTemplate {
```

Comment:
You could make `ApplyTemplate` a case class, and do away with all the `*Creator` classes. They seem unnecessary.